c10/util/env.h

```diff
@@ -14,7 +14,14 @@ namespace utils {
 // NB:
 // Issues a warning if the value of the environment variable is not 0 or 1.
 inline optional<bool> check_env(const char* name) {
+#ifdef _MSC_VER
```

Comment:
`torch_test_cpp_extension` does some compilation at testing time https://github.com/pytorch/pytorch/actions/runs/3890569745/jobs/6640722956 and it fails with the following error.

```
C:\actions-runner\_work\pytorch\pytorch\build\win_tmp\build\torch\include\c10/util/env.h(17): error C4996: 'getenv': This function or variable may be unsafe. Consider using _dupenv_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
```

This looks unrelated to the linker issue, so I suspect that a recent commit merged while Windows CPU build and tests are disable causes this ( no signal).  The fix looks simple enough though.

Reply:
JFYI, `getenv_s` should be used instead https://en.cppreference.com/w/c/program/getenv

Reply:
Noted, I'll fix this properly in another PR and will merge this once Windows CPU build and test pass.

Reply:
Alternatively including `<cstdlib>` together with the `std::` namespace clasifier might resolve the issue as https://stackoverflow.com/a/631696/685292 or other sources mentions, though I haven't tried it.